sklearn/cluster/k_medoids_.py

```diff
@@ -0,0 +1,279 @@
+# -*- coding: utf-8 -*-
+"""K-medoids clustering"""
+
+# Authors: Timo Erkkilä <timo.erkkila@gmail.com>
+#          Antti Lehmussola <antti.lehmussola@gmail.com>
+#          Kornel Kiełczewski <kornel.k@plusnet.pl>
+# License: BSD 3 clause
+
+import warnings
+
+import numpy as np
+
+from ..base import BaseEstimator, ClusterMixin, TransformerMixin
+from ..metrics.pairwise import pairwise_distances
+from ..utils import check_array, check_random_state
+from ..utils.validation import check_is_fitted
+
+
+class KMedoids(BaseEstimator, ClusterMixin, TransformerMixin):
+    """k-medoids class.
+
+    Read more in the :ref:`User Guide <k_medoids>`.
+
+    Parameters
+    ----------
+    n_clusters : int, optional, default: 8
+        How many medoids. Must be positive.
+
+    distance_metric : string, optional, default: 'euclidean'
+        What distance metric to use.
```

Comment:
That's correct.. however, I'm wondering what I should put here, since the best would be to copy the description from pairwise.pairwise_distances

```Valid values for metric are: ... ``` 

and/or 

```    metric : string, or callable
        The metric to use when calculating distance between instances in a
        feature array. If metric is a string...
```

Maybe I could reference the method pairwise_distance somehow (using a ref tag)? This is effectively what KMedoids uses internally.